cpp/src/arrow/flight/sql/example/sqlite_server.cc

```diff
@@ -0,0 +1,684 @@
+    table_query << " and schema_name LIKE '" << command.schema_filter_pattern.value()
+                << "'";
+  }
+
+  if (command.table_name_filter_pattern.has_value()) {
+    table_query << " and table_name LIKE '" << command.table_name_filter_pattern.value()
+                << "'";
+  }
+
+  if (!command.table_types.empty()) {
+    table_query << " and table_type IN (";
+    size_t size = command.table_types.size();
+    for (size_t i = 0; i < size; i++) {
+      table_query << "'" << command.table_types[i] << "'";
+      if (size - 1 != i) {
+        table_query << ",";
+      }
+    }
+
+    table_query << ")";
+  }
+
+  table_query << " order by table_name";
+  return table_query.str();
+}
+
+Status SetParametersOnSQLiteStatement(sqlite3_stmt* stmt, FlightMessageReader* reader) {
+  FlightStreamChunk chunk;
+  while (true) {
+    RETURN_NOT_OK(reader->Next(&chunk));
+    std::shared_ptr<RecordBatch>& record_batch = chunk.data;
+    if (record_batch == nullptr) break;
+
+    const int64_t num_rows = record_batch->num_rows();
+    const int& num_columns = record_batch->num_columns();
+
+    for (int i = 0; i < num_rows; ++i) {
+      for (int c = 0; c < num_columns; ++c) {
+        const std::shared_ptr<Array>& column = record_batch->column(c);
+        ARROW_ASSIGN_OR_RAISE(std::shared_ptr<Scalar> scalar, column->GetScalar(i));
+
+        auto& holder = static_cast<DenseUnionScalar&>(*scalar).value;
+
+        switch (holder->type->id()) {
+          case Type::INT64: {
+            int64_t value = static_cast<Int64Scalar&>(*holder).value;
+            sqlite3_bind_int64(stmt, c + 1, value);
+            break;
+          }
+          case Type::FLOAT: {
+            double value = static_cast<FloatScalar&>(*holder).value;
+            sqlite3_bind_double(stmt, c + 1, value);
+            break;
+          }
+          case Type::STRING: {
+            std::shared_ptr<Buffer> buffer = static_cast<StringScalar&>(*holder).value;
+            sqlite3_bind_text(stmt, c + 1, reinterpret_cast<const char*>(buffer->data()),
+                              static_cast<int>(buffer->size()), SQLITE_TRANSIENT);
+            break;
+          }
+          case Type::BINARY: {
+            std::shared_ptr<Buffer> buffer = static_cast<BinaryScalar&>(*holder).value;
+            sqlite3_bind_blob(stmt, c + 1, buffer->data(),
+                              static_cast<int>(buffer->size()), SQLITE_TRANSIENT);
+            break;
+          }
+          default:
+            return Status::Invalid("Received unsupported data type: ",
+                                   holder->type->ToString());
+        }
+      }
+    }
+  }
+
+  return Status::OK();
+}
+
+}  // namespace
+
+std::shared_ptr<DataType> GetArrowType(const char* sqlite_type) {
+  if (sqlite_type == NULLPTR) {
+    // SQLite may not know the column type yet.
+    return null();
+  }
+
+  if (boost::iequals(sqlite_type, "int") || boost::iequals(sqlite_type, "integer")) {
```

Comment:
Why use `boost::iequals`? Is it actually possible to get different casings here?

Reply:
Yes, `sqlite_type` here comes exactly as written in a `CREATE TABLE` statement.